pkg/datapath/loader/base.go

```diff
@@ -424,6 +543,9 @@ func (l *Loader) Reinitialize(ctx context.Context, o datapath.BaseProgramOwner,
 		args[initArgProxyRule] = "false"
 	}
 
+	// Required when cilium is running on Kind
+	args[initArgNetNsCookie] = getNetNsCookie(ctx)
```

Comment:
I discussed the overall approach with @brb a bit, and the problem I currently see in here is that it's racy. Meaning, whenever we Reinitialize() (one example would be agent up/down-grade), we shortly swap the bind() progs from the existing socket LB with the cookie probe ones. While they do SYS_REJECT, my concern is that it will lead to hard to debug application issues in production (why does my app occasionally return -EPERM on bind?). Our thinking is we expose the netns cookie via getsockopt similar to the socket cookie and route it as a stable fix. Once that is complete, we can just allocate a socket from the agent in getNetNsCookie(), do the getsockopt() call, and are done with it.

Reply:
> While they do SYS_REJECT, my concern is that it will lead to hard to debug application issues in production (why does my app occasionally return -EPERM on bind?).

That's an amazing catch! I agree ;)

> Our thinking is we expose the netns cookie via getsockopt similar to the socket cookie and route it as a stable fix. Once that is complete, we can just allocate a socket from the agent in getNetNsCookie(), do the getsockopt() call, and are done with it.

So, the plan could be to add a `BPF_PROG_TYPE_CGROUP_SOCKOPT` (on bpf_sock.c, for exemple) which ensures an unused argument (assuming `SO_NETNS_COOKIE=-1`, for exemple) and (maybe) `EINVAL` as syscall ret. And that, under these circumstances, set `optval=netns`?

Or would it be better wait for linux kernel to support netns cookie?
Thanks for the discussion!